beater/cloudbeat.go

```diff
@@ -132,6 +134,12 @@ func (bt *cloudbeat) Run(b *beat.Beat) error {
 		select {
 		case <-bt.ctx.Done():
 			return nil
+
+		case update := <-bt.configUpdates:
```

Comment:
Maybe this is enough for now, but I think theres a design problem with having this channel watched here.
I'm thinking of the next step - updating the bundle we use OPA to eval with.
The evaluator was already created, as part of `New`, As were the fetchers and the manager.
How will we change their config from this scope? Shouldn't we have some sort of cloudbeat factory here that will restart everything with the new config?
Let me know what you think

Reply:
I agree with @eyalkraft  on this one. If we created a new cloudbeat for every reconfigure it would make a lot of sense.
Though there is a product impact on such a change:
After reconfiguring cloudbeat, when do we run the next fetchers iteration?
1. Based on the old configuration timeout
2. Based on the new configuration timeout
3. Right away, and then - based on the new configuration timeout

Creating a new cloudbeat instance probably means we choose (3) which I think is fine, just need to make sure that is accepted by everyone.

Reply:
Thanks for your comments @eyalkraft, @amirbenun 

You're correct; other beats have different ways of implementing what ultimately becomes a restart of the `beater` interface. As discussed in our sync, this is just a fast, hacky way to get to our goals for 8.3. I considered going for the big architecture change, and we do want to get there eventually, but it didn't seem safe at this time partly because we don't have a truly "end to end" integration testing framework yet.

I've confirmed that this PR as it is serves our purpose for now. It's not a PR to reconfigure the beat, although it takes a step in that direction. It's just a PR to use config updates to get the `dataYaml`, while entirely ignoring all the other contents of the config update.